taichi/ir/control_flow_graph.cpp

```diff
@@ -644,6 +642,8 @@ void ControlFlowGraph::reaching_definition_analysis(bool after_lower_access) {
     to_visit.push(nodes[i].get());
     in_queue[nodes[i].get()] = true;
   }
+
+  // The worklist algorithm.
   while (!to_visit.empty()) {
```

Comment:
Can we implement "the worklist algo" as a separate function?

Reply:
(I thought I replied by email 7 days ago but it didn't appear here) Probably no — IIRC the reaching definition analysis and the live variable analysis use opposite (forward/backward) edges in the graph.